re-write dropdown for a better UX

leejordan opened the pull request:
features:

- adds dropdown.js for progressive enhancement
- adds a delay on mouseenter and mouseleave to avoid people switching tabs when trying to move to an item in the dropdown (via use of css transitions)
- enables it to work via keyboard or controller via use of focus states
- enables use on touch screens via handling the `click` and `touchstart` events instead of just using css hover as before
- when a dropdown tab is enabled via `click` or `touchstart` a secondary click anywhere else in the document will close any open dropdowns
- adds the ability to provide a data attribute `data-dropdown-offscreen="xxs,xs"` which determines the breakpoints at which the dropdown ceases to act like a dropdown and instead like an accordion
- no longer use the accordion module to make it behave like an accordion when in the off screen menu, use some js in the dropdown module instead
- a better BEM structure
- more refined css overrrides for when dropdown is offscreen

@notlee needs testing across devices - should also incidentally fix the old bug you found with the accordion in the off screen menu on ios4

notlee commented:
caught by copy and paste `/layouts/drop-down-menu`, unnecessary "Shop Categories" header.
![screen shot 2016-02-08 at 17 20 24](https://cloud.githubusercontent.com/assets/10405691/12893320/562bad60-ce88-11e5-9846-0475ce8bbeee.png)


The focus state strikes me as broken whilst hovering over other nav elements, as discussed, maybe we can remove the focus state when hovering over different elements?
![screen shot 2016-02-08 at 17 19 06](https://cloud.githubusercontent.com/assets/10405691/12893348/7e69d84c-ce88-11e5-90a6-d41776dcfccd.png)

notlee commented on js/src/modules/drop-down.js:
```diff
@@ -0,0 +1,134 @@
+        }
+
+        function clickOffScreen(e) {
+            console.log('click');
```
erroneous log

notlee commented:
The page becomes horizontally scrollable around portrait tablet size with a white bar to the right. I haven't figured out exactly why but at a glance it seems to be to do with `.nav__dropdown__menu`, i.e. applying `overflow: hidden;` works as a quick fix.

![screen shot 2016-02-09 at 10 58 11](https://cloud.githubusercontent.com/assets/10405691/12914634/6327f542-cf1d-11e5-8e3d-1831018a4947.png)
![screen shot 2016-02-09 at 10 52 03](https://cloud.githubusercontent.com/assets/10405691/12914633/6320db04-cf1d-11e5-88e9-174ea324745d.png)

notlee commented:
I still feel a little iffy about the way behaviour is split within the drop down to support the offscreen-menu, but seen as they are the same elements just with changed behaviour in different viewport contexts it does make some sense and provides the appropriate level of flexibility for Pistachio. :+1:

Otherwise looks good! I'll give this some more testing after you've seen the above bug. :)

leejordan commented:
@notlee I have addressed your comments (apart from the unhelpful nebulous complaint where you describe my life's work as "iffy")

Please merge if you're happy.

notlee commented:
@leejordan Good stuff. This is adequate. :wink: Merging.
